docs/porting/target/tickless/tickless.md

```diff
@@ -0,0 +1,64 @@
+
+![](./resources/Tick.png)
+
+Tickless mode is an optimization mechanism to suspend the SysTick timer. In particular, this will put the system to uninterrupted sleep to achieve power saving when idle for multiple ticks. Target in tickless mode normally disables the SysTick, sets up wake-up timers and enters sleep mode when idle. When SysTick timer expires on its own or from an external event like interrupt, then it exits sleep mode and re-enables the SysTick.
+
+[Mbed OS sleep modes](../../../apis/power-management-sleep.html):
+
+- Sleep mode: Both low and high-frequency clocks are active.
+- Deep sleep mode: Only the low-frequency clock is active.
+
+To enable tickless mode in Mbed OS, your target has to meet following requirements:
+
+- Support for sleep HAL API
+- Support for low power or microsecond ticker HAL API
+
+If the target supports both low power and microsecond ticker, then low power ticker is selected with tickless mode.
+
+For instance, the system runs only with two threads thread 1 and thread 2. During the first 2 tick interrupts thread 1 and thread 2 schedules alternatively. When thread 1 entered into a wait state, the scheduler preempts and schedules thread 2 for execution. After some time, thread 2 enters into a wait state and get preempted. Now scheduler schedules idle thread to run as two threads are in the wait state. This idle thread put the system to sleep. After multiple SysTick periodic interrupts, thread 2 is rescheduled for execution. All previous Systick wake-ups waste power.
+
+![](./resources/Normal_Tick.png)
+
+But with tickless mode, it suspends the SysTick periodic wake up as no threads are ready to run for quite some time which inturn saves the power.
+
+![](./resources/Tickless.png)
+
+### Enable tickless mode support in full profile
+
+Find your target in `mbed-os/targets/targets.json` and update its configuration as described below.
+
+1. Add `MBED_TICKLESS` macro into the `macros` option as follows:
+
+    ```json
+	"macros": [
+        "MBED_TICKLESS"
+    ]    
+    ```
+1. If your target does not have a low power ticker or you prefer to use microsecond ticker instead of low power ticker for tickless mode, then ensure that tickless is enabled using the microsecond ticker:
+
+    ```json
+    "overrides": {
+        "tickless-from-us-ticker": true
+    }
+    ```
+
+Targets supporting tickless mode override the default SysTick mechanism and use the [OsTimer](../mbed-os-api-doxy/structos__timer__def.html) implementation based on either the [low power ticker](../mbed-os-api-doxy/group__hal__lp__ticker.html) or [microsecond ticker](../mbed-os-api-doxy/group__hal__us__ticker.html) if "tickless-from-us-ticker" target configuration is set. This change is necessary to avoid drift connected with using two different timers to measure time. Users must not change the ticker interrupt handler when tickless mode is in use.
```

Comment:
Can you provide more explanation to `Users must not change the ticker interrupt handler when tickless mode is in use.`?